### PR TITLE
lang/funcs: update cidrsubnet and cidrhost for 64-bit systems

### LINKED ISSUE #25360: cidrsubnet doesn't work with ipv6  subnets
### Terraform Version

```
Terraform v0.12.3
+ provider.packet v2.9.0

Your version of Terraform is out of date! The latest version
is 0.12.26. You can update by downloading from www.terraform.io/downloads.html

```

### Terraform Configuration Files

```hcl
locals {
service_cidr_range = cidrsubnet("fe08::/64", 63, 0)
}
```

### Expected Behavior
https://www.terraform.io/docs/configuration/functions/cidrsubnet.html documents `cidrsubnet` works for ipv6. and subnet sizes can go upto `128`.  So expected output here is
```
service_cidr_range => "fe08::/127"
```

### Actual Behavior
```

Error: Error in function call

  on provision.tf line 46, in locals:
  46:   service_cidr_range = cidrsubnet("fe08::/64", 63, 0)

Call to function "cidrsubnet" failed: may not extend prefix by more than 32
bits.
```

### Steps to Reproduce
```
terraform apply
```
### Additional Context
none

### References
none

### PR DESCRIPTION
**Pull request by mildwonkey**
The `cidrsubnet` and `cidrhost` functions were limited to supporting 32-bit systems. This PR:
* updates the `"github.com/apparentlymart/go-cidr"` library, which was recently extended with SubnetBig and HostBig functions
* switches to the new go-cidr functions, including passing appropriate values as `big.Int`s
* modifies terraform's validation check in `cidrsubnet` to check for bits > 64 instead of 32

Fixes  #25360

There's a new test, but for good measure, have some `terraform console` output:
```
> cidrsubnet("fe08::/64", 63, 0)
fe08::/127
```

### CODE REVIEW

File: lang/funcs/cidr.go [hunk 1]
```diff
@@ -25,7 +26,7 @@ var CidrHostFunc = function.New(&function.Spec{
 	},
 	Type: function.StaticReturnType(cty.String),
 	Impl: func(args []cty.Value, retType cty.Type) (ret cty.Value, err error) {
-		var hostNum int
+		var hostNum *big.Int
```
Comment (apparentlymart) [inline lang/funcs/cidr.go:29]: I had honestly completely forgotten that `gocty` knows how to decode into `big.Int`! I'm glad to remember now that it does, cause I was originally expecting this change to be a bit more invasive.

One thing I notice looking back at the test cases for these functions is that they aren't testing the situation where the `newbits`, `hostnum`, `netnum` arguments are fractional numbers. I'm expecting that all of those would be properly rejected as invalid by the `gocty` conversions here (it should say something like "value must be a whole number") but might be worth adding some "expected error" test cases as insurance to future changes that might inadvertently make those panic otherwise.

File: lang/funcs/cidr.go [hunk 2]
```diff
@@ -96,15 +97,11 @@ var CidrSubnetFunc = function.New(&function.Spec{
 			return cty.UnknownVal(cty.String), fmt.Errorf("invalid CIDR expression: %s", err)
 		}
 
-		// For portability with 32-bit systems where the subnet number
-		// will be a 32-bit int, we only allow extension of 32 bits in
-		// one call even if we're running on a 64-bit machine.
-		// (Of course, this is significant only for IPv6.)
-		if newbits > 32 {
-			return cty.UnknownVal(cty.String), fmt.Errorf("may not extend prefix by more than 32 bits")
+		if newbits > 64 {
```
Comment (apparentlymart) [inline lang/funcs/cidr.go:100]: I don't think we need to have this check at all anymore, because the underlying CIDR library will already check to make sure that there's enough room to extend by `newbits` bits. I had originally put the `> 32` check here just to make sure this function wouldn't behave differently on 32-bit vs. 64-bit builds of Terraform, but using `*big.Int` now means that the behavior of `SubnetBig` should be the same for all architectures and so we shouldn't need to apply additional constraints in the Terraform layer here.

(When using IPv6, it would also be "valid" to start off with a very short prefix like e.g. `1::/1` and extend that by up to 127 new bits, which `SubnetBig` should accept just fine now and might be worth including a test for just to exercise the boundary cases a bit more.)

File: lang/funcs/cidr.go (hunk 1 above)
Comment (mildwonkey) [reply lang/funcs/cidr.go:29]: Done and done!
